verify/config_test.go

```diff
@@ -211,30 +210,40 @@ func (matcher *ValidJSONMatcher) NegatedFailureMessage(actual interface{}) (mess
 	return format.Message(actual, "to _not_ be valid JSON")
 }
 
-//func skipIfVmtype(vmType define.VMType, message string) {
-//	if isVmtype(vmType) {
-//		Skip(message)
-//	}
-//}
-//
-//func skipIfNotVmtype(vmType define.VMType, message string) {
-//	if !isVmtype(vmType) {
-//		Skip(message)
-//	}
-//}
-//
-//func skipIfWSL(message string) {
-//	skipIfVmtype(define.WSLVirt, message)
-//}
-//
-//func isVmtype(vmType define.VMType) bool {
-//	return testProvider.VMType() == vmType
-//}
-//
-//// isWSL is a simple wrapper to determine if the testprovider is WSL
-//func isWSL() bool {
-//	return isVmtype(define.WSLVirt)
-//}
+func (m *imageTestBuilder) skipIfVmtype(vmType string, message string) {
+	if m.isVmtype(vmType) {
+		Skip(message)
+	}
+}
+
+func (m *imageTestBuilder) skipIfNotVmtype(vmType string, message string) {
+	if !m.isVmtype(vmType) {
+		Skip(message)
+	}
+}
+
+const (
+	QemuVirt = "wsl"
+	WSLVirt = "qemu"
+	AppleHvVirt = "applehv"
+	HyperVVirt = "hyperv"
+	LibKrun = "libkrun"
+	UnknownVirt = "unknown"
+)
+
+func (m *imageTestBuilder) isVmtype(vmType string) bool {
+	return vmType == m.vmType()
+}
+
+func (m *imageTestBuilder) vmType() string {
```

Comment:
I will merge it for now but this really should be cached once at the beginning as we do quite a few calls.